crates/typed-store/src/metrics.rs

```diff
@@ -172,3 +319,22 @@ impl DBMetrics {
         }
     }
 }
+
+#[derive(Debug)]
+pub struct DBMetrics {
+    pub op_metrics: OperationMetrics,
+    pub cf_metrics: ColumnFamilyMetrics,
+}
+
+impl DBMetrics {
+    pub(crate) fn new(registry: &Registry) -> Self {
+        DBMetrics {
+            op_metrics: OperationMetrics::new(registry),
+            cf_metrics: ColumnFamilyMetrics::new(registry),
+        }
+    }
+    pub fn make_db_metrics(registry: &Registry) -> &'static Arc<DBMetrics> {
+        static ONCE: OnceCell<Arc<DBMetrics>> = OnceCell::new();
+        ONCE.get_or_init(|| Arc::new(DBMetrics::new(registry)))
+    }
```

Comment:
This is going to present a problem since what happens if i call `make_db_metrics` twice with two different registries? (We will likely do this in a number of places). Please re-think this, or remove it.

Reply:
It is a todo to find a better solution for this (unrelated to this PR btw). If we did not create a static instance, we will end up reinitializing these prometheus metrics every time we create a new DBMap which prometheus does not like and complains. The right way to fix this is to create an instance of Arc<DBMetrics> in sui initialization code and pass it down everywhere DBMap is being created through some kind of macro based options. We do only use one registry to report metrics in a sui instance though, no?

Reply:
Yes we only use one registry.  At least for Sui.   Narwhal code used its own registry, but I'm not sure about now now that it's integrated (Narwhal has its own metrics port too, again if it was one registry it would be integrated - @bmwill do you know?)